embedchain/loaders/github.py

```diff
@@ -2,116 +2,258 @@
+            for subtree in tree:
+                if subtree.type == "tree":
+                    _get_repo_tree_contents(repo_path, subtree, progress_bar)
+                else:
+                    assert subtree.type == "blob"
                     try:
-                        results = future.result()
-                        if results:
-                            data.extend(results)
-                            data_urls.extend([result.get("meta_data").get("url") for result in results])
-                    except Exception as e:
-                        logging.warn(f"Failed to process {file}: {e}")
-
-        source_hash = hashlib.sha256(repo_url.encode()).hexdigest()
-        repo_path = f"/tmp/{source_hash}"
-        _fetch_or_clone_repo(repo_url=repo_url, local_path=repo_path)
-        _add_repo_files(repo_path)
-        doc_id = hashlib.sha256((repo_url + ", ".join(data_urls)).encode()).hexdigest()
+                        contents = subtree.data_stream.read().decode("utf-8")
+                    except Exception:
+                        logging.warning(f"Failed to read file: {subtree.path}")
+                        progress_bar.update(1) if progress_bar else None
+                        continue
+
+                    url = f"{repo_url.rstrip('.git')}/blob/main/{subtree.path}"
+                    data.append(
+                        {
+                            "content": clean_string(contents),
+                            "meta_data": {
+                                "url": url,
+                            },
+                        }
+                    )
+                if progress_bar is not None:
+                    progress_bar.update(1)
+
+        repo_tree = _get_repo_tree(repo_url, local_path)
+        tree_list = list(repo_tree.traverse())
+        with tqdm(total=len(tree_list), desc="Loading files:", unit="item") as progress_bar:
+            _get_repo_tree_contents(local_path, repo_tree, progress_bar)
+
+        return data
+
+    def _github_search_repo(self, query: str):
+        """Search github repo."""
+        data = []
+        logging.info(f"Searching github repos with query: {query}")
+        results = self.client.search_repositories(query)
+        # Add repo urls and descriptions
+        urls = list(map(lambda x: x.html_url, results))
+        discriptions = list(map(lambda x: x.description, results))
+        data.append(
+            {
+                "content": clean_string(desc),
+                "meta_data": {
+                    "url": url,
+                },
+            }
+            for url, desc in zip(urls, discriptions)
+        )
+
+        # Add repo contents
+        for result in results:
+            clone_url = result.clone_url
+            print("Cloning repository: ", clone_url)
+            data = self._get_github_repo_data(clone_url)
+        return data
+
+    def _github_search(self, query: str, type: str):
```

Comment:
Can we please change the function name to something like `_github_issues_search()`? The current name is quite ambiguous.